Update the existing "Settings" screen to match the new design

anikiki opened the pull request:
<!--
Note: This checklist is a reminder of our shared engineering expectations. 
The items in Bold are required
If your PR involves UI changes:
    1. Upload screenshots or screencasts that illustrate the changes before / after
    2. Add them under the UI changes section (feel free to add more columns if needed)
    3. Make sure these changes are tested in API 23 and API 26
If your PR does not involve UI changes, you can remove the **UI changes** section
-->

Task/Issue URL: https://app.asana.com/0/0/1204667877992124/f

### Description
Updated the Settings screen.

### Steps to test this PR

_Settings - Privacy section_

- [ ] Install from this branch and go to the `Settings` screen.
- [ ] Notice the items are the same as in Design (the sections / order / copy): Default Browser, Private Search, Web Tracking Protection, Cookie Pop-Up Protection, AppTP, NetP.
- [ ] Tap on `Default Browser` and notice the screen to choose the default browser is opened (as in Production).
- [ ] Go back to the Settings screen. 
- [ ] Tap on `Private Search` and notice the new Private Search screen looks as in design (icon and copy at the top, Search Settings title, Autocomplete Suggestions, More Search Settings). 
- [ ] Go back to the Settings screen. 
- [ ] Tap on `Web Tracking Protection` and notice the new Web Tracking Protection screen looks as in design (icon and copy at the top, GPC and Unprotected Sites).
- [ ] Go back to the Settings screen. 
- [ ] Tap on `Cookie Pop-Up Protection` and notice the new Cookie Pop-Up Protection screen looks as in design (copy at the top, Let DDG handle cookie pop-ups toggle). 
- [ ] Go back to the Settings screen. 
- [ ] Tap on `Email Protection` and notice you are sent to the browser tab (as in Production).
- [ ] Go back to the Settings screen. 
- [ ] Tap on `App Tracking Protection` and notice AppTP onboarding is opened (as in Production).
- [ ] Go back to the Settings screen. 
- [ ] Tap on `Network Protection` and notice the waitlist screen is opened (as in Internal).
- [ ] Go back to the Settings screen. 


_Settings - Settings section_

- [ ] Tap on `Autofill Logins` and notice Logins is opened (as in Production).
- [ ] Go back to the Settings screen. 
- [ ] Tap on `Sync` and notice Sync is opened (as in Internal).
- [ ] Go back to the Settings screen. 
- [ ] Tap on `Home Screen Widget` and notice the add widget prompt (as in Production).
- [ ] Tap on `Fire Button` and notice the new Fire Button screen looks as in design (Fireproof Sites, Automatically Clear, Clear On, Fire Button Animation).
- [ ] Go back to the Settings screen. 
- [ ] Tap on `Permissions` and notice the new Permissions screen looks as in design (Site Permissions, Notifications, Open Links in Apps).
- [ ] Go back to the Settings screen. 
- [ ] Tap on `Appearance` and notice the new Appearance looks as in design (Theme, App Icon).
- [ ] Go back to the Settings screen. 
- [ ] Tap on `Accessibility` and notice the Accessibility is opened (as in Production).
- [ ] Tap on `About` and notice the updated About is opened (icon and copy at the top, Privacy Policy, Version, Share Feedback).
- [ ] Go back to the Settings screen. 
- [ ] Tap on `DuckDuckGo Mac App` and notice the DuckDuckGo Mac screen is opened (as in Production).
- [ ] Go back to the Settings screen. 
- [ ] Tap on `DuckDuckGo Windows App` and notice the DuckDuckGo Windows screen is opened (as in Production).
- [ ] Go back to the Settings screen. 
- [ ] All internal features as unchanged.


### UI changes
See https://app.asana.com/0/0/1204969193599390/f

anikiki commented:
Current dependencies on/for this PR:
* develop
  * **PR #3226** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3226" target="_blank"><img src="https://static.graphite.dev/graphite-32x32.png" alt="Graphite" width="10px" height="10px"/></a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3226?utm_source=stack-comment).

malmstein reviewed (commented):
A few nits and questions:
- Did you consider using the new Nav library instead?
- Are some pixels still missing?

anikiki commented on app/src/main/java/com/duckduckgo/app/about/AboutDuckDuckGoViewModel.kt:
```diff
@@ -0,0 +1,129 @@
+        return command.receiveAsFlow()
+    }
+
+    fun onStartActivityCalled() {
```
Thanks, updated in the other PR.

anikiki commented:
> A few nits and questions:
> 
>     * Did you consider using the new Nav library instead?
> 
>     * Are some pixels still missing?

Updated in the other PR to use the Nav library. Added all pixels.

malmstein commented on app/src/main/res/layout/activity_private_search.xml:
```diff
@@ -0,0 +1,103 @@
+                android:id="@+id/privateSearchAutocompleteToggle"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_marginTop="16dp"
```
nit: 
```suggestion
                android:layout_marginTop=“@dimen/keyline_4"
```

malmstein reviewed (approved):
Great work, I left just a suggestion for a dimension. Works as expected.
